### PR TITLE
refactor(sidepanel): change slug to aiLabel

### LINKED ISSUE #6095: SidePanel: `slug` --> `aiLabel` renaming
```[tasklist]
### Tasks
- [ ] Change slug prop to aiLabel
- [ ] Deprecate slug prop
- [ ] Add new duplicate classname selectors to the ailabel.scss styles
- [ ] Update classnames within AILabel components to use the new #{$prefix}--ailabel names
- [ ] Create a new sub-issue/tasklist item on https://github.com/carbon-design-system/ibm-products/issues for removing these classes as part of new major version
```

For duplicate class, this is an example:
```
- .#{$prefix}--slug .#{$prefix}--slug__button {

+ .#{$prefix}--slug .#{$prefix}--slug__button,
+ .#{$prefix}--ai-label .#{$prefix}--ai-label__button {
```

Image

### PR DESCRIPTION
**Pull request by makafsal**
Closes #6095 #6119 

#### What did you change?
`slug` to `aiLabel`

#### How did you test and verify your work?
storybook

### CODE REVIEW

File: packages/ibm-products-styles/src/components/SidePanel/_side-panel.scss [hunk 1]
```diff
@@ -109,7 +109,7 @@ $action-set-block-class: #{c4p-settings.$pkg-prefix}--action-set;
     left: 0;
     border-right: 1px solid $border-subtle-02;
   }
-  &.#{$block-class}.#{$block-class}--has-slug {
+  &.#{$block-class}.#{$block-class}--has-ai-label {
```
Comment (elycheea) [inline packages/ibm-products-styles/src/components/SidePanel/_side-panel.scss:112]: I believe the decision was to include _duplicate_ selectors[^1] to be backwards compatible.

```suggestion
  &.#{$block-class}.#{$block-class}--has-slug,
  &.#{$block-class}.#{$block-class}--has-ai-label {
```

[^1]: https://github.com/carbon-design-system/carbon/issues/17714

**Review by elycheea — CHANGES_REQUESTED** [review]
Instead of replacing `slug`, I believe the cross-squad decision was to duplicate classnames rather than only replacing them.

File: packages/ibm-products-styles/src/components/SidePanel/_side-panel.scss [hunk 2]
```diff
@@ -464,7 +469,10 @@ $action-set-block-class: #{c4p-settings.$pkg-prefix}--action-set;
   inset: 0;
 }
 
-.#{$block-class}--has-slug + .#{$block-class}__overlay {
+.#{$block-class}--has-slug
+  + .#{$block-class}__overlay
```
Comment (devadula-nandan) [inline packages/ibm-products-styles/src/components/SidePanel/_side-panel.scss:473]:
```suggestion
  + .#{$block-class}__overlay,
```

File: packages/ibm-products/src/components/SidePanel/SidePanel.stories.jsx [hunk 3]
```diff
@@ -219,9 +219,14 @@ const actionSets = [
...
 ];
 
-const sampleSlug = (
-  <Slug className="slug-container" size="xs">
-    <SlugContent>
+const sampleAILabel = (
+  <AILabel
+    className="aiLabel-container"
+    size="xs"
+    autoAlign={false}
```
Comment (devadula-nandan) [inline packages/ibm-products/src/components/SidePanel/SidePanel.stories.jsx:226]: i think we dont need to pass this false explicitly.
```suggestion
```

**Review by devadula-nandan — COMMENTED** [review]
2 comments, rest looks good to me.

**Review by devadula-nandan — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Review by elycheea — APPROVED** [review]

**Pull request merged**